### DISCUSSION TITLE
Stringtie executables is in PATH variable

### DISCUSSION THREAD

**Discussion by selenaajohn** [nextflow-io/nextflow#3775] [Q&A] [answered]
I'm trying to use stringtie in my nextflow pipeline and I am getting an error saying: 
<img width="428" alt="Screen Shot 2023-03-19 at 4 35 37 PM" src="https://user-images.githubusercontent.com/97220964/226214022-1f2d2ab8-3e03-434d-924b-27f5d932856e.png">
My issue is that I haven't properly ensured that that the path with newly-built executables is in PATH variable. I have tried adding something to my nextflow conf.ig file, but it hasn't worked. I also tried adding stringtier = "/data/selena/stringtie-2.2.1" to the nextflow config file

nextflow config file:
<img width="660" alt="Screen Shot 2023-03-19 at 4 28 58 PM" src="https://user-images.githubusercontent.com/97220964/226213843-24493abf-c9ae-4c23-8e3e-7a0a7351c2ef.png">

Nextlow pipeline:
<img width="213" alt="Screen Shot 2023-03-19 at 4 29 31 PM" src="https://user-images.githubusercontent.com/97220964/226213816-36c219aa-5df4-4356-a658-009e897e4ea6.png">
<img width="598" alt="Screen Shot 2023-03-19 at 4 29 38 PM" src="https://user-images.githubusercontent.com/97220964/226213799-1dc263c6-f287-40e8-91a1-03748524a96f.png">

Thanks.

**Comment by mribeirodantas** [comment, marked as answer]
Do:

```
env {
  PATH = $PATH:/data/selena/stringtie-2.2.1
}
```
instead. Or, you can also move the stringtie binary to the `.bin/` folder within the project folder of your pipeline. Nextflow will automatically add it to the `$PATH` of the task. Also, ideally, you should use a container image with the tools you need.

**Reply by selenaajohn** [reply]
Would that be in the config file or the pipeline?

Thanks for the tip!

**Reply by mribeirodantas** [reply]
In the `nextflow.config` file (or any other config file correctly provided, with `-c/-C`, for example). You're welcome 😄
